### DISCUSSION TITLE
Editing the DSL for an on-premise installation

### DISCUSSION THREAD

**Discussion by caroe233** [structurizr/cli#120] [Q&A]
Hi all,

I am currently trying to establish an on-premise installation for my team. It is now running and I identified that editing the DSL directly seems not to work for on-premise, so my understanding is I need to use the CLI for editing. Is this correct? And if so, why can I only pull the json and not the dsl file? How can I then edit the dsl file to push it back? (always having in mind that the next edit might be done by another colleague...)

Would be very grateful if someone could help me understanding the idea of the concept / workflow here...

**Comment by simonbrowndotje** [comment]
> and I identified that editing the DSL directly seems not to work for on-premise

The DSL editor is disabled by default; see https://structurizr.com/share/18571/documentation#structurizrproperties

I would recommend joining the [Slack group](https://join.slack.com/t/structurizr/shared_invite/enQtMzkyMjY1NzMwNTkzLTcyOGI1MTZmNDQwMDQ5YmZlMThiYmU1ZTM2ZWZiMzYwMjVhNmM0OWIwNjFlZTM1YmY3YzU0ZDY2MTA1YTk5Mjg) and asking others about their workflows.
